torch/fx/experimental/graph_gradual_typechecker.py

```diff
@@ -106,6 +106,13 @@ def add_inference_rule(n: Node):
         return n.type
 
     (new_t1, new_t2) = broadcast_types(t1, t2)
+
+    if new_t1 != t1:
+        n.args[0].broadcast = True  # type:ignore[attr-defined]
```

Comment:
are `n.args[0]` and `args[1]` expected to always be `Node`s? Would probably be a good idea to add an assert to check that and then put the broacast flag in `node.meta` rather than monkey-patching a new attribute

Reply:
I fixed it and also the broadcasting part we talked about which was in the previous code. Let me know it looks reasonable!